plugin-kbindicator/src/kbdstateconfig.cpp

```diff
@@ -112,5 +112,5 @@ void KbdStateConfig::save()
 
 void KbdStateConfig::configureLayouts()
 {
-    QProcess::startDetached(QL1S("lxqt-config-input --show-page \"Keyboard Layout\""));
+    QProcess::startDetached(QL1S("lxqt-config-input"), QStringList() << QL1S("show-page \"Keyboard Layout\""));
```

Comment:
I think (and tested) that `--` shouldn't be dropped.

Reply:
Oh, sorry, I was wrong about `--`. I tested it with another app, not "lxqt-config-input"; should find out when it's needed and when not...

Reply:
OK, found it. I was both right and wrong ;) Only this works fine:

```c++
QProcess::startDetached(QL1S("lxqt-config-input"), QStringList() << QL1S("--show-page") << QL1S("Keyboard Layout"));
```

Reply:
Will take a more thorough look.
It didn't work with the two dashes.

Reply:
I hope the following explanation will be helpful (it's also about parsing):

`QProcess` needs all arguments as they are and separate from each other. We don't need to worry about spaces and quotes anymore. On the one hand, *with input commands*, we should see if there are arguments, e.g., in this way (adapted from FeatherPad's code):

```c++
QString command = input_command;
auto commandParts = command.split (QRegularExpression ("\\s+"), Qt::SkipEmptyParts);
if (!commandParts.isEmpty())
{
    command = commandParts.takeAt (0); // there may be arguments
    process->start (command, QStringList() << commandParts);
}
else // the input was empty
...
```

I got a little confused because I had followed the above rule successfully in FeatherPad but saw a contradiction here.

Reply:
> It didn't work with the two dashes

I made the same mistake in my second comment ;) It shouldn't work unless you separate "--show-page" from "Keyboard Layout" — which should be done.

Reply:
wouldn't be better something like:
replace:
`QProcess::startDetached(exec);`

with:
```
#if (QT_VERSION >= QT_VERSION_CHECK(5,15,0))
    QStringList args = QProcess::splitCommand(exec);
    QProcess::startDetached(args.takeFirst(), args);
#else
    QProcess::startDetached(exec);
#endif
```